### PR TITLE
Encode / Decode color images

### PR DESCRIPTION
**Pull request by trstephen**
`wm_encode` and `wm_decode` support grayscale and color images 🌈 

Also did PSNR comparisons for embedding the watermark in each layer of RGB, YCbCr. Surprisingly, embedding in RGB gives better PSNR than YCbCr!

```
Embedding "the best watermark for peppers" in peppers.tiff variants

      source             wm image           psnr   
---------------------------------------------------
peppersYCbCr.tiff    peppersY_wm.tiff     58.91533
peppersYCbCr.tiff    peppersCb_wm.tiff    58.12800
peppersYCbCr.tiff    peppersCr_wm.tiff    58.95000
peppersRGB.tiff      peppersR_wm.tiff     63.60164
peppersRGB.tiff      peppersG_wm.tiff     63.74127
peppersRGB.tiff      peppersB_wm.tiff     63.70763


Embedding "fruits watermark okay!" in fruits.tiff variants

      source              wm image          psnr   
---------------------------------------------------
fruitsYCbCr.tiff     fruitsY_wm.tiff      58.88691
fruitsYCbCr.tiff     fruitsCb_wm.tiff     57.95389
fruitsYCbCr.tiff     fruitsCr_wm.tiff     59.07317
fruitsRGB.tiff       fruitsR_wm.tiff      63.88637
fruitsRGB.tiff       fruitsG_wm.tiff      63.69708
fruitsRGB.tiff       fruitsB_wm.tiff      63.59464
```

This test can be re-run with `color_psnr_comparison.m`. Just need to supply an image with an RGB and YCbCr variants. This can be created with the popular [convert tool from imagemagick](http://www.imagemagick.org/script/index.php). Use the command:
```
convert SomeRGB.tiff -colorspace ycbcr -compress none SomeYCbCr.tiff
```

### CODE REVIEW

File: color_psnr_comparison.m [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+% Now using the watermark added DCT coefficients, construct and output the
+% watermarked image as Watermarked.tiff
+im_Y_idct  = blockproc(im_Y_dct,  [8 8], @(b) idct2(b.data));
+im_Cb_idct = blockproc(im_Cb_dct, [8 8], @(b) idct2(b.data));
+im_Cr_idct = blockproc(im_Cr_dct, [8 8], @(b) idct2(b.data));
+im_R_idct  = blockproc(im_R_dct,  [8 8], @(b) idct2(b.data));
+im_G_idct  = blockproc(im_G_dct,  [8 8], @(b) idct2(b.data));
+im_B_idct  = blockproc(im_B_dct,  [8 8], @(b) idct2(b.data));
+
+% Copy original source file to preserve all tiff metadata. It's a pain to write by hand..
```
Comment (houtan1) [inline color_psnr_comparison.m:115]: This section is interesting

**Review by houtan1 — APPROVED** [review]
Wow Tyler, looks really cool! I'll merge your PR.

File: color_psnr_comparison.m (hunk 1 above)
Comment (trstephen) [reply color_psnr_comparison.m:115]: Inspired by [this](https://www.mathworks.com/matlabcentral/answers/218829-transferring-tiff-image-tags-identically-to-another-image)

Comment (trstephen) [reply color_psnr_comparison.m:115]: The Matlab folks, uh, expect you to write this all by hand

[link](https://www.mathworks.com/help/matlab/import_export/exporting-to-images.html)

**Pull request merged**